File: src/main/scala/com/vc/medium/L413.scala
package com.vc.medium

object L413 {
  def numberOfArithmeticSlices(arr: Array[Int]): Int = {
    val n = arr.length
    var cur = 0
    var sum = 0
    (2 until n).foreach(i => {
      if(arr(i - 2) - arr(i - 1) == arr(i - 1) - arr(i)) {
        cur += 1
        sum += cur
      }
      else cur = 0
    })
    sum
  }
}

File: src/main/java/com/vc/medium/L413.java
package com.vc.medium;

class L413 {
    public int numberOfArithmeticSlices(int[] A) {
        if(A == null || A.length == 0) return 0;

        int n = A.length, total = 0;
        int[] dp = new int[n];
        for(int i = 2; i < n; i++) {
            if(A[i - 1] - A[i - 2] == A[i] - A[i - 1]) {
                dp[i] = dp[i - 1] + 1;
                total += dp[i];
            }
        }
        //System.out.println(Arrays.toString(dp));
        return total;
    }
}
